### PR TITLE
complete jumpcloud

### PR DESCRIPTION
**Pull request by mbliakharskyi**
## Description

Please include a summary of the changes and issues addressed in this PR. Also, explain the motivation and context of each update. List any dependencies or prerequisites needed for implementing this change.

## Checklist:

Before you create this PR, confirm all the requirements listed below by checking the checkboxes like this (`[x]`).

- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added tests to cover the new feature or fixes.

### CODE REVIEW

File: apps/jumpcloud/src/app/install/service.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  region: string;
+};
+
+export const registerOrganisation = async ({
+  organisationId,
+  apiKey,
+  region,
+}: SetupOrganisationParams) => {
+  const encodedApiKey = await encrypt(apiKey);
+
```
Comment (elbalexandre) [inline apps/jumpcloud/src/app/install/service.ts:18]: We should do a dumb call to check apiKey validity like `await getUsers(...)`

File: apps/jumpcloud/src/connectors/users.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+  const response = await fetch(url.toString(), {
+    method: 'DELETE',
+    headers: {
+      'Content-Type': 'application/json',
+      'x-api-key': `${apiKey}`,
+    },
+  });
+
+  if (!response.ok && response.status !== 404 && response.status !== 400) {
```
Comment (elbalexandre) [inline apps/jumpcloud/src/connectors/users.ts:88]: Is it 404 or 400 when a user does not exists ?

File: apps/jumpcloud/src/inngest/client.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      data: {
+        organisationId: string;
+        region: string;
+      };
+    };
+    'jumpcloud/users.delete.requested': {
+      data: {
+        userId: string;
+      };
+    };
```
Comment (elbalexandre) [inline apps/jumpcloud/src/inngest/client.ts:33]: We need to pass organisationId (that can be retrieved with `parseWebhookEvent`)
```suggestion
    'jumpcloud/users.delete.requested': {
      data: {
        userId: string;
        organisationId: string
      };
    };
```

File: apps/jumpcloud/src/inngest/functions/users/delete-users.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  { id: 'delete-users' },
+  { event: 'jumpcloud/users.delete.requested' },
+  async ({ event }) => {
+    const { userId } = event.data;
+
+    const [organisation] = await db
+      .select({
+        apiKey: Organisation.apiKey,
+      })
+      .from(Organisation);
```
Comment (elbalexandre) [inline apps/jumpcloud/src/inngest/functions/users/delete-users.ts:18]: I miss that in the previous PR review. But, we should select the org using id retrieved from event data.
```suggestion
      .from(Organisation)
      .where(eq(Organisation.id, organisationId));
```

File: apps/jumpcloud/src/connectors/users.ts (hunk 2 above)
Comment (mbliakharskyi) [reply apps/jumpcloud/src/connectors/users.ts:88]: I tested it.
1. 404 error: invalid user_id
2. 400 error: user not exist.

Comment (mbliakharskyi) [reply apps/jumpcloud/src/connectors/users.ts:88]: > I tested it.
> 
> 1. 400 error: invalid user_id
> 2. 404 error: user not exist.

Comment (mbliakharskyi) [reply apps/jumpcloud/src/connectors/users.ts:88]: image

Comment (mbliakharskyi) [reply apps/jumpcloud/src/connectors/users.ts:88]: image

Comment (elbalexandre) [reply apps/jumpcloud/src/connectors/users.ts:88]: Ok so let's drop the 400 error filter, it's ok to throw in this case.

File: apps/jumpcloud/.env.test [hunk 5]
```diff
@@ -0,0 +1,20 @@
+ELBA_SOURCE_ID="0442f541-45d2-487a-9e4b-de03ce4c559e"
+ELBA_API_KEY="abc123dfg456"
+ELBA_API_BASE_URL="http://127.0.0.1:3522/api/rest"
+ELBA_REDIRECT_URL="https://admin.elba.jumplogic/install"
+ELBA_WEBHOOK_SECRET="123"
+
+JUMPCLOUD_API_BASE_URL="https://api-sandbox.jumplogic.app/"
```
Comment (elbalexandre) [inline apps/jumpcloud/.env.test:7]: We should use V2 api if possible

Comment (mbliakharskyi) [reply apps/jumpcloud/.env.test:7]: let me check

**Pull request merged**